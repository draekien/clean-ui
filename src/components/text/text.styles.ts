import { ThemeUIStyleObject } from 'theme-ui';
import { HorizontalAlignment } from '../../types/layouts';
import { colors } from '../theme/theme.colors';
import { textVariants } from '../theme/theme.variants';

export interface textCssProps {
  variant: keyof typeof textVariants;
  color: keyof typeof colors | 'inherit' | 'initial';
  textAlign: HorizontalAlignment;
  fullWidth: boolean;
  truncate: boolean;
}

export const textCss = ({
  variant,
  color,
  textAlign,
  truncate,
  fullWidth,
}: textCssProps): ThemeUIStyleObject => {
  const css: any = {
    variant: `text.${variant}`,
    color: color,
    display: fullWidth ? 'block' : 'inline-block',
    textAlign: textAlign,
    transition: 'all 300ms',
  };

  if (truncate) {
    css.overflow = 'hidden';
    css.textOverflow = 'ellipsis';
    css.whiteSpace = 'nowrap';
    css.display = 'inline-block';
    css.maxWidth = '100%';
  }

  return css;
};
