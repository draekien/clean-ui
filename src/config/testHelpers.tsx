import React, { ReactNode } from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: ReactNode) =>
  render(<ThemeProvider>{element}</ThemeProvider>);
