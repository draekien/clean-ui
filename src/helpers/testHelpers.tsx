import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: React.ReactNode) => {
  return render(<ThemeProvider>{element}</ThemeProvider>);
};

export const hexToRgb = (hex: string) => {
  hex = hex.replace(/[^0-9A-F]/gi, '');
  const bigint = parseInt(hex, 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgb(${r}, ${g}, ${b})`;
};
