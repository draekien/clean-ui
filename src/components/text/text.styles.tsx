import { ThemeUIStyleObject } from 'theme-ui';
import { colors } from '../theme/theme.colors';
import { TextVariant } from './text';

export interface textCssProps {
  variant: TextVariant;
  color: keyof typeof colors;
}

export const textCss = ({ variant, color }: textCssProps): ThemeUIStyleObject => {
  const css: any = {
    variant: `text.${variant}`,
    color: color,
  };

  return css;
};
