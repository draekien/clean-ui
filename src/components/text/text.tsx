/** @jsxImportSource theme-ui */
import React from 'react';
import { Text as ThemeUiText } from 'theme-ui';
import { colors } from '../theme/theme.colors';
import { textVariants } from '../theme/theme.variants';
import { HorizontalAlignment } from '../../types/alignments';
import * as styles from './text.styles';

export interface TextProps {
  /** The type of DOM element to render
   * @default 'div'
   */
  as?: React.ElementType<any>;
  /** The text variant
   * @default 'body'
   */
  variant?: keyof typeof textVariants;
  /** The color of the text
   * @default 'text'
   */
  color?: keyof typeof colors;
  /** The text to render */
  children: React.ReactNode;
  /** horizontal alignment of the text
   * @default 'left'
   */
  textAlign?: HorizontalAlignment;
  /** should text take up the full width of the parent container */
  fullWidth?: boolean;
  /** should text be truncated instead of overflow */
  truncate?: boolean;
}

export const Text: React.FC<TextProps> = ({
  as,
  children,
  variant = 'body',
  color = 'text',
  textAlign = 'left',
  truncate = false,
  fullWidth = false,
}) => {
  return (
    <ThemeUiText
      as={as}
      sx={styles.textCss({ variant, color, textAlign, truncate, fullWidth })}>
      {children}
    </ThemeUiText>
  );
};
