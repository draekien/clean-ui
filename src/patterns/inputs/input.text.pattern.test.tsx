import React from 'react';
import { themedRender } from '../../helpers/testHelpers';
import { TextInput } from './input.text.pattern';

describe('Text input pattern', () => {
  it('should match snapshot', () => {
    const { container } = themedRender(<TextInput inputId="text" label="text" />);

    expect(container).toMatchSnapshot();
  });
});
