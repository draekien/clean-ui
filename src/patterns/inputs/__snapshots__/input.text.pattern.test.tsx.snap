// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text input pattern should match snapshot 1`] = `
<div>
  <div
    class="css-109th3q-InputWrapper"
  >
    <div
      class="css-1qm0f6g-InputLabel"
    >
      <label
        for="text"
      >
        <div
          class="css-14i3diu-Text css-vurnku"
        >
          text
        </div>
      </label>
    </div>
    <div
      class="css-xcsx4z"
    >
      <input
        class="css-msl6v2"
        id="text"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
